reliquery/storage.py

```diff
@@ -436,6 +437,158 @@ def get_all_relic_data(self) -> List[Dict]:
+        name: str,
+    ):
+        self.access_token = access_token
+        self.dbx = dropbox.Dropbox(access_token)
+        self.prefix = "/" + prefix
+        self.name = name
+
+    def _join_path(self, path: StoragePath) -> str:
+        return "/".join([self.prefix] + path)
+
+    def put_file(self, path: StoragePath, file_path: str) -> None:
+        path = path.split()
+        with open(file_path, "rb") as f:
+            self.dbx.files_upload(f.read(), "/" + self._join_path(path), mute=True)
+
+    def put_binary_obj(self, path: StoragePath, buffer: BytesIO):
+        self.dbx.files_upload(buffer.read(), self._join_path(path), mute=True)
+
+    def get_binary_obj(self, path: StoragePath) -> BytesIO:
+        return BytesIO(
+            self.dbx.files_download(self._join_path(path), rev=None)[-1].content
+        )
+
+    def put_text(self, path: StoragePath, text: str) -> None:
+        string_bytes = bytes(text, "utf-8")
+        self.dbx.files_upload(string_bytes, self._join_path(path))
+
+    def get_text(self, path: StoragePath) -> str:
+        try:
+            return self.dbx.files_download(self._join_path(path))[-1].text
+        except ApiError:
+            raise StorageItemDoesNotExist
+
+    def list_keys(self, path: StoragePath) -> List[str]:
+        try:
+            return [
+                i.name
+                for i in self.dbx.files_list_folder(self._join_path(path)).entries
+            ]
+        except ApiError:
+            return []
+
+    def put_metadata(self, path: StoragePath, metadata: Dict):
+        metadata_bytes = json.dumps(metadata).encode("utf-8")
+        self.dbx.files_upload(
+            metadata_bytes,
+            self._join_path(path),
+            mode=dropbox.files.WriteMode.overwrite,
+        )
+
+    def get_metadata(self, path: StoragePath, root_key: str) -> Dict:
+        dirs = [
+            "arrays",
+            "html",
+            "text",
+            "images",
+            "json",
+            "pandasdf",
+            "files",
+            "notebooks",
+        ]
+
+        data = {
+            root_key: {
+                "arrays": [],
+                "text": [],
+                "html": [],
+                "images": [],
+                "json": [],
+                "pandasdf": [],
+                "files": [],
+                "notebooks": [],
+            }
+        }
+
+        def dict_from_path(path: StoragePath, dirname: str):
+            dirpath = path.copy()
+            dirpath.append(dirname)
+            entries = self.list_keys(dirpath)
+
+            for i in entries:
+                entry_path = dirpath.copy()
+                entry_path.append(i)
+                data[root_key][dirname].append(
+                    self.dbx.files_download(self._join_path(entry_path))[-1].json()
+                )
+
+        for d in dirs:
+            dict_from_path(path, d)
+
+        return data
+
+    # def get_all_relic_metadata(self) -> List[Dict]:
```

Comment:
I would search for anything calling this method. if nothing does then it is safe to remove it.